# DP: readline63-005 upstream patch

			   READLINE PATCH REPORT
			   =====================

Readline-Release: 6.3
Patch-ID: readline63-005

Bug-Reported-by:	Juergen Daubert <jue@jue.li>
Bug-Reference-ID:	<20140303180430.GA7346@jue.netz>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-readline/2014-03/msg00002.html

Bug-Description:

There are still applications using the deprecated Function/VFunction/etc.
typedefs in rltypedefs.h.  This patch restores the typedefs, but attempts
to mark them as deprecated using gcc/clang attributes.  Thanks to Max Horn
for the suggestion.

Patch (apply with `patch -p0'):

Index: b/rltypedefs.h
===================================================================
--- a/rltypedefs.h
+++ b/rltypedefs.h
@@ -26,6 +26,25 @@
 extern "C" {
 #endif
 
+/* Old-style, attempt to mark as deprecated in some way people will notice. */
+
+#if !defined (_FUNCTION_DEF)
+#  define _FUNCTION_DEF
+
+#if defined(__GNUC__) || defined(__clang__)
+typedef int Function () __attribute__ ((deprecated));
+typedef void VFunction () __attribute__ ((deprecated));
+typedef char *CPFunction () __attribute__ ((deprecated));
+typedef char **CPPFunction () __attribute__ ((deprecated));
+#else
+typedef int Function ();
+typedef void VFunction ();
+typedef char *CPFunction ();
+typedef char **CPPFunction ();
+#endif
+
+#endif /* _FUNCTION_DEF */
+
 /* New style. */
 
 #if !defined (_RL_FUNCTION_TYPEDEF)
Index: b/patchlevel
===================================================================
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-4
+5
